### ISSUE TITLE
I wonder what setting did you use when reporting the DAVIS+YouTubeVOS only(training data) score on YouTubeVOS2018 validation set?

### ISSUE THREAD

**Issue by Shallow2000** [hkchengrex/XMem#32]
Thanks for your wonderful work!  Did you use the setting of stage 2 only when reporting  the DAVIS+YouTubeVOS only(training data) score on YouTubeVOS2018 validation set? I retrained stage 2 but it only gains 82.6 on YouTubeVOS2018 validation set with no extra change.

**Comment by hkchengrex** [comment]
Yes, it is trained on stage 2 only. No other things were changed. Even training with stage 3 (shorter training) should give decent results (~84.0). 82.6 is far too low and I think something is wrong. Can you evaluate your model on DAVIS 2017 validation and see how it performs?

**Comment by Shallow2000** [comment]
The result on DAVIS 2017 validation is the same as yours (84.4 vs 84.5).

**Comment by hkchengrex** [comment]
Hmm... Are you using the migrated evaluation server? Can you submit our precomputed results to this new server? One of my speculations is that the new server is doing something different and does not produce the same result.

**Comment by Shallow2000** [comment]
Yes, I used the migrated YouTubeVOS 2018 evaluation server that was updated yesterday. I will try to submit the precomputed results to this new server soon and compare it with the result of old server. Thank you again!

**Comment by Shallow2000** [comment]
Umm... I just test three precomputed results and the evaluation scores on the new server are the same as the older one (some subtle difference because of precision of machine). Maybe it is not about the migration of evaluation server.

**Comment by Shallow2000** [comment]
What is the torch and cuda version you used? Can it be caused by the difference of environment?

**Comment by hkchengrex** [comment]
I think it was PyTorch 1.11. CUDA 10 and 11 perform the same. To control that, do you mind using our pretrained weights (not results) to evaluate on your machine and submit the results?

**Comment by Shallow2000** [comment]
I have used your pretrained weights to evaluate on my machine and the results are reported as the same of your paper. Therefore, I guess the exception happens in the training process.

**Comment by Shallow2000** [comment]
My environment is PyTorch 1.12 and CUDA 11.3. And I retrained stage 2 directly.

**Comment by hkchengrex** [comment]
I see. It is pretty surprising that the DAVIS performance matches with a large gap in YouTubeVOS... Have you tried looking at the last few saved weights but not just the final one? Usually the difference there is only around 0.1~0.3 but you might have stumbled upon a particularly bad local minima.

**Comment by Shallow2000** [comment]
I tried the last 5 saved weights with the gap of 5k iterations. And I retrained for 2 times. The results shows the same pattern and I am pretty confused about it.

**Comment by hkchengrex** [comment]
Yeah that's confusing. Let me investigate a bit. Thank you for reporting.

**Comment by Shallow2000** [comment]
Thank you for your patient analysis and wonderful help!

**Comment by hkchengrex** [comment]
I just retrained two new models with Pytorch 1.12 and 1.11 respectively. Both use 4 V100s and run on CUDA 10.2 with driver version 440.64. Without model selection (I just picked the 157K model), their scores are 84.2 and 84.4 respectively. I cannot reproduce your results and therefore cannot continue the investigation. More details to follow.

![Screenshot 2022-09-29 171912](https://user-images.githubusercontent.com/7107196/193153553-6ffe5adb-a245-4978-af5e-3955ca599af4.jpg)
(I don't know why the lower scoring result is put on the leaderboard)

**Comment by hkchengrex** [comment]
The training commands are:
```
CUDA_VISIBLE_DEVICES=0,1,2,3 OMP_NUM_THREADS=4 torchrun --master_port 56785 --nproc_per_node=4 train.py --stage 2 --benchmark --exp_id xmem-s2-retrain
```

```
CUDA_VISIBLE_DEVICES=4,5,6,7 OMP_NUM_THREADS=4 torchrun --master_port 56786 --nproc_per_node=4 train.py --stage 2 --benchmark --exp_id xmem-s2-retrain-pytorch1.11
```

The evaluation commands are:
```
CUDA_VISIBLE_DEVICES=0 python eval.py  --output ../output/yv18-Sep28/xmem-s2-retrain-107K --model saves/Sep28_12.08.42_xmem-s2-retrain_s2/Sep28_12.08.42_xmem-s2-retrain_s2_157000.pth
```

```
CUDA_VISIBLE_DEVICES=1 python eval.py  --output ../output/yv18-Sep28/xmem-s2-retrain-pytorch1.11-107K --model saves/Sep28_12.14.26_xmem-s2-retrain-pytorch1.11_s2/Sep28_12.14.26_xmem-s2-retrain-pytorch1.11_s2_157000.pth --dataset Y18
```

**Comment by hkchengrex** [comment]
The environment (with `conda list`): [list.txt](https://github.com/hkchengrex/XMem/files/9679440/list.txt)

**Comment by hkchengrex** [comment]
Tensorboard log (of the PyTorch 1.11 training): https://drive.google.com/file/d/1QBeOF3kWrg4vm6AOJ3X8ICuLcpfDXuEA/view?usp=sharing

**Comment by Shallow2000** [comment]
Thank you very much! I just found my problem and I am pretty sorry to bother you. When I evaluated the checkpoint on the new machine I did not download the "all_frames" dataset folder and 
![image](https://user-images.githubusercontent.com/106326662/193412205-6bac50e7-f695-4032-8f1e-5aac6ea5151e.png)
changed the self.image_dir in test_datasets.py to path.join(data_root, split, 'JPEGImages'). That is how the problem is caused. I got 84.0 score when the problem is solved.
